deepchecks/nlp/context.py

```diff
@@ -216,18 +216,47 @@ def _validate_classification_prediction(dataset: TextData, prediction: TTextPred
     @staticmethod
     def _validate_token_classification_prediction(dataset: TextData, prediction: TTextPred):
         """Validate prediction for given token classification dataset."""
-        if not all(isinstance(pred, collections.abc.Sequence) for pred in prediction):
-            raise ValidationError(f'Check requires predictions for {dataset.name} to be a sequence '
-                                  f'of sequences')
-
-        for i in range(len(prediction)):  # TODO: Goes over all predictions, fix this
-            if not all(isinstance(pred, str) for pred in prediction[i]) \
-                    and not all(isinstance(pred, int) for pred in prediction[i]):
-                raise ValidationError(f'Check requires predictions for {dataset.name} to be a sequence '
-                                      f'of sequences of strings or integers')
-            if len(prediction[i]) != len(dataset.tokenized_text[i]):
-                raise ValidationError(f'Check requires predictions for {dataset.name} to have '
-                                      f'the same number of tokens as the input text')
+        if (
+            isinstance(prediction, (str, bytes, bytearray))
+            or not isinstance(prediction, (np.ndarray, t.Sequence))
+            # TODO: fix error messages, make it more particular
+            # third party containers do not extend Sequence. Example:
```

Comment:
agree - see is_sequence_not_str function as alternative

Reply:
In general the NLP dummy model is not that good IMO (in the future it should be updated or removed - see https://linear.app/deepchecks/issue/DEE-456/unify-dummy-models)

Reply:
replaced with `is_sequence_not_str`